devops/scripts/sycl-bisect.bash

```diff
@@ -0,0 +1,205 @@
+                                  in the current working directory
+  -b,--command-allow-bisect-codes Pass 125 and >127 return codes to git-bisect
+                                  directly when using --command to allow the
+                                  command to skip commits or stop bisection. If
+                                  this option is not present, any non-zero
+                                  return value is interpreted as a bad commit.
+
+Examples:
+
+"Clang :: SemaSYCL/accessor_inheritance.cpp" started failing on sycl-web because
+of one of the commits merged by 96f730774ac4. To find which one of these 96
+commits caused this test failure, sycl-bisect.bash can be run like this:
+
+$ devops/scripts/sycl-bisect.bash 96f730774ac4 96f730774ac4^ --test clang/test/SemaSYCL/accessor_inheritance.cpp
+HELP
+      exit 0
+      ;;
+    '-t'|'--test')
+      export TEST="$2"
+      shift 2
+      ;;
+    '-c'|'--command')
+      export COMMAND="$2"
+      shift 2
+      ;;
+    '-b'|'--command-allow-bisect-codes')
+      export COMMAND_ALLOW_BISECT_CODES=1
+      shift
+      ;;
+    '--')
+      shift
+      break
+      ;;
+    *)
+      echo "Unexpected argument: $1" >&2
+      exit 1
+      ;;
+  esac
+done
+
+# Validate options.
+if [[ "$1" == "" ]] || [[ "$2" == "" ]]; then
+  echo "A bad rev and at least one good rev must be passed" >&2
+  exit 1
+fi
+if [[ "$TEST" == "" ]] && [[ "$COMMAND" == "" ]]; then
+  echo "--test <test> or --command <command> must be used to specify a test" >&2
+  exit 1
+fi
+if [[ "$TEST" != "" ]] && [[ "$COMMAND" != "" ]]; then
+  echo "Only one of --test <test> or --command <command> is allowed" >&2
+  exit 1
+fi
+if [[ "$COMMAND_ALLOW_BISECT_CODES" != "" ]] && [[ "$COMMAND" == "" ]]; then
+  echo "--command-allow-bisect-codes is only allowed with --command <command>" >&2
+  exit 1
+fi
+
+# Save the current working directory before switching to the repository's
+# top-level directory. Tests will be run relative to the original working
+# directory.
+export TEST_WD="$PWD"
+TOPLEVEL="$(git rev-parse --show-toplevel)"
+if [[ "$TOPLEVEL" == "" ]]; then
+  exit 1
+fi
+cd "$TOPLEVEL"
+
+# A commit that should be on all sycl-based branches this script works with.
+# This is set to the initial sycl-specific commit.
+export SYCL_ANCESTOR=1e0b4966ba9a
+
+# This commit should be in the repository.
+if ! git rev-parse $SYCL_ANCESTOR &>/dev/null; then
+  echo "Current repository is not sycl-based" >&2
+  exit 1
+fi
+
+# Along with all the bad/good revs.
+for REV in "$@"; do
+  if ! git rev-parse --verify "$REV" &>/dev/null; then
+    echo "'$REV' is not a valid revision" >&2
+    exit 1
+  fi
+done
+
+# Make sure the build directory is set up; if not, tell the user to run
+# the configure step.
+if [[ ! -f build/CMakeCache.txt ]]; then
+  echo "The build directory doesn't seem to be configured yet." >&2
+  echo "Please run the configure step to set up the build." >&2
+  exit 1
```

Comment:
To make this message more you useful, you can consider adding a reference to the GetStartedGuide, where we have up-to-date valid configuration instructions.

E.g. https://intel.github.io/llvm-docs/GetStartedGuide.html#build-dpc-toolchain or https://github.com/intel/llvm/blob/sycl/sycl/doc/GetStartedGuide.md#build-dpc-toolchain

Reply:
I've updated this message to reference the documentation